src/main/java/com/jwplayer/southpaw/Southpaw.java

```diff
@@ -482,77 +360,65 @@ protected DenormalizedRecord createDenormalizedRecord(
+                metrics.denormalizedRecordsCreatedByTopicAndPriority.get(root.getDenormalizedName()).get(priority).mark(1);
+                createdDePKs.add(dePrimaryKey);
+            }
+            if(config.createRecordsTimeS > 0 && timer.getTime() > config.createRecordsTimeS * 1000L) {
+                break;
             }
-            metrics.denormalizedRecordsCreated.mark(1);
-            metrics.denormalizedRecordsCreatedByTopic.get(root.getDenormalizedName()).mark(1);
-            metrics.denormalizedRecordsToCreate.update(metrics.denormalizedRecordsToCreate.getValue() - 1);
-            metrics.denormalizedRecordsToCreateByTopic.get(root.getDenormalizedName())
-                    .update(metrics.denormalizedRecordsToCreateByTopic.get(root.getDenormalizedName()).getValue() - 1);
         }
+        if(createdDePKs.size() == rootRecordPKs.size()) {
+            rootRecordPKs.clear();
+        } else {
+            rootRecordPKs.removeAll(createdDePKs);
+        }
+        handleTimers();
     }
 
     /**
-     * Create the entry name for the denormalized PKs yet to be created
-     * @return - The entry name
-     */
-    protected String createDePKEntryName(Relation root) {
-        return String.join(SEP, PK, root.getDenormalizedName());
-    }
-
-    /**
-     * Simple class for creating a FK multi index
-     * @param indexName - The name of the index to create
-     * @param indexedTopicName - The name of the indexed topic
-     * @return A brand new, shiny index
-     */
-    protected BaseIndex<BaseRecord, BaseRecord, Set<ByteArray>> createFkIndex(
-            String indexName,
-            String indexedTopicName) {
-        MultiIndex<BaseRecord, BaseRecord> index = new MultiIndex<>();
-        index.configure(indexName, rawConfig, state, inputTopics.get(indexedTopicName));
-        return index;
-    }
-
-    /**
-     * Creates all indices for all relations provided to Southpaw. Note: indices to the input records
-     * can be shared between top level relations.
+     * Create the keyspace for the primary keys of the denormalized records yet to be created. Only low and high
+     * priorities have the priority in the keyspace for backwards compatibility.
+     * @param root - The root relation
+     * @param priority - The priority of the records to be created
+     * @return - The name of the keyspace
      */
-    protected void createIndices() {
-        for(Relation root: relations) {
-            // Children - PK, parent key and join key indices for the input topics
-            for(Relation child: root.getChildren()) {
-                createChildIndices(root, root, child);
-            }
+    protected static String createRecordsToBeCreatedKeyspace(Relation root, QueueingStrategy.Priority priority) {
+        if(priority == QueueingStrategy.Priority.MEDIUM) {
+            return String.join(SEP, PK, root.getDenormalizedName());
```

Comment:
Why not having the `MEDIUM` priority following the same pattern as others?

Reply:
For backwards compatibility. The medium queue is the old queue. Only the low and high ones are new.